File: src/test/java/com/binance/api/examples/MarketDataEndpointsExample.scala
package com.binance.api.examples

import com.binance.api.client.BinanceApiClientFactory
import com.binance.api.client.domain.Symbol
import com.binance.api.client.domain.market.CandlestickInterval.WEEKLY
import com.binance.api.client.exception.BinanceApiExceptionError

import scala.concurrent.ExecutionContext.Implicits.global

/**
  * Examples on how to get market data information such as the latest price of a symbol, etc.
  */
object MarketDataEndpointsExample extends App {

  val factory = new BinanceApiClientFactory("", "")
  val client  = factory.newAsyncRestClient
  val symbol  = Symbol("NEOETH")

  // Getting depth of a symbol
  client.getOrderBook(symbol, 10).foreach((orderBook) => println(orderBook.asks))
  // Getting latest price of a symbol
  client.get24HrPriceStatistics(symbol).foreach(println)
  // Getting all latest prices
  client.getAllPrices.foreach(println)
  // Getting agg trades
  client.getAggTrades(symbol, limit = Some(10)).foreach(println)
  // Weekly candlestick bars for a symbol
  client.getCandlestickBars(symbol, WEEKLY).foreach(println)
  // Getting all book tickers
  client.getBookTickers.foreach(println)
  // Exception handling
  client.getOrderBook(Symbol("UNKNOWN"), 10).failed.foreach {
    case e: BinanceApiExceptionError =>
      println(e.error.code) // -1121
      println(e.error.msg)  // Invalid symbol
    case th => th.printStackTrace()
  }

}

File: src/test/java/com/binance/api/examples/MarketDataEndpointsExample.java
package com.binance.api.examples;

import com.binance.api.client.BinanceApiAsyncRestClient;
import com.binance.api.client.BinanceApiClientFactory;
import com.binance.api.client.exception.BinanceApiException;

import java.util.function.Consumer;

import static com.binance.api.client.domain.market.CandlestickInterval.WEEKLY;
import static java.util.Optional.empty;

/**
 * Examples on how to get market data information such as the latest price of a symbol, etc.
 */
public class MarketDataEndpointsExample {

    public static void main(String[] args) {
        BinanceApiClientFactory factory = BinanceApiClientFactory.newInstance();
        BinanceApiAsyncRestClient client = factory.newAsyncRestClient();
        Consumer<Object> print = System.out::println;

        // Getting depth of a symbol
        client.getOrderBook("NEOETH", 10).thenAccept(orderBook -> System.out.println(orderBook.asks));

        // Getting latest price of a symbol
        client.get24HrPriceStatistics("NEOETH").thenAccept(print);

        // Getting all latest prices
        client.getAllPrices().thenAccept(print);

        // Getting agg trades
        client.getAggTrades("NEOETH", empty(), empty(), empty(), empty()).thenAccept(print);

        // Weekly candlestick bars for a symbol
        client.getCandlestickBars("NEOETH", WEEKLY, empty(), empty(), empty()).thenAccept(print);

        // Getting all book tickers
        client.getBookTickers().thenAccept(print);

        // Exception handling
        client.getOrderBook("UNKNOWN", 10).handle((orderBook, th) -> {
            if (th instanceof BinanceApiException) {
                BinanceApiException e = (BinanceApiException) th;
                System.out.println(e.getError().getCode()); // -1121
                System.out.println(e.getError().getMsg());  // Invalid symbol
            }
            return null;
        });
    }
}
